lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,287 @@
+            return []
+        }
+
+        // Validate the body of the description function.
+        //   1. The body must have a single item
+        //   2. The single item must be a string literal
+        //   3. Later on, the interpolation in the string literal will be validated.
+        guard let codeBlock = descriptionProperty.accessorBlock?.accessors.as(CodeBlockItemListSyntax.self),
+              let descriptionString = codeBlock.single?.item.as(StringLiteralExprSyntax.self) else {
+            let message: ErrorMessage = "\(bindingName(descriptionProperty)!) must consist of a single string literal"
+            context.diagnose(node: descriptionProperty, message: message)
+            return []
+        }
+
+        // Precompute which properties are known to be computed. Used for producing diagnostics.
+        var computedProperties: Set<String> = []
+        for property in properties.values {
+            if isComputedProperty(property), let name = bindingName(property) {
+                computedProperties.insert(name)
+            }
+        }
+
+        // Iterate the string's segments, and convert property expressions into LLDB variable references.
+        var summarySegments: [String] = []
+        for segment in descriptionString.segments {
+            switch segment {
+            case let .stringSegment(segment):
+                summarySegments.append(segment.content.text)
+            case let .expressionSegment(segment):
+                guard let labeledExpr = segment.expressions.single, labeledExpr.label == nil else {
+                    // This catches `appendInterpolation` overrides.
+                    let message: ErrorMessage = "unsupported custom string interpolation expression"
+                    context.diagnose(node: segment, message: message)
+                    return []
+                }
+
+                let expr = labeledExpr.expression
+
+                var propertyChain: [String] = []
+                if let declRef = expr.as(DeclReferenceExprSyntax.self) {
+                    // A reference to a single property on self.
+                    propertyChain.append(declRef.baseName.text)
+                } else if let memberAccess = expr.as(MemberAccessExprSyntax.self) {
+                    // A chain of properties, possibly starting with an explicit self.
+                    enumerateMembers(memberAccess) { declRef in
+                        propertyChain.append(declRef.baseName.text)
+                    }
+                } else {
+                    // The expression was neither a DeclReference nor a MemberAccess.
+                    let message: ErrorMessage = "unsupported expression; stored properties only"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                // Explicit self are removed before use by LLDB.
+                if propertyChain[0] == "self" {
```

Comment:
propertyChain might be empty:
```suggestion
                if propertyChain.first == "self" {
```

Reply:
Oh never mind, didn't see the early return